packages/datetime/src/timePicker.tsx

```diff
@@ -163,15 +174,22 @@ export class TimePicker extends React.Component<ITimePickerProps, ITimePickerSta
         const didMinTimeChange = nextProps.minTime !== this.props.minTime;
         const didMaxTimeChange = nextProps.maxTime !== this.props.maxTime;
         const didBoundsChange = didMinTimeChange || didMaxTimeChange;
+        const didUseAmPmChange = nextProps.useAmPm !== this.props.useAmPm;
+
+        let value = this.state.value;
+        let useAmPm = this.props.useAmPm;
 
         if (didBoundsChange) {
-            const timeInRange = DateUtils.getTimeInRange(this.state.value, nextProps.minTime, nextProps.maxTime);
-            this.setState(this.getFullStateFromValue(timeInRange));
+            value = DateUtils.getTimeInRange(this.state.value, nextProps.minTime, nextProps.maxTime);
         }
-
         if (nextProps.value != null && !DateUtils.areSameTime(nextProps.value, this.props.value)) {
-            this.setState(this.getFullStateFromValue(nextProps.value));
+            value = nextProps.value;
+        }
+        if (didUseAmPmChange) {
```

Comment:
i think this check is useless and it is equivalent to just passing through `nextProps.useAmPm`. if props are unchanged then next is the same as `this.props`. please remove (just like we did on line 130).